src/interpolators.jl

```diff
@@ -128,6 +133,85 @@ function (sim::ScatteredInterpolationMethod)(
+
+See the [NaturalNeighbours documentation](https://github.com/DanielVandH/NaturalNeighbours.jl) for more details.
+
+This method is fully configurable and will forward all arguments to the relevant NaturalNeighbours.jl functions.
+To understand what the methods do, see the documentation for [`NaturalNeighbours.jl`](https://github.com/DanielVandH/NaturalNeighbours.jl).
+
+## Keyword arguments
+
+The main keyword argument to look at here is `method`, which is the method
+to use for the interpolation.  The other keyword arguments are simply forwarded 
+to NaturalNeighbours.jl's interpolator.
+
+$(Makie.DocStringExtensions.FIELDS)
+"""
+@with_kw struct NaturalNeighboursMethod <: Interpolator
+    "The method to use for the interpolation.  Defaults to `Sibson(1)."
+    method::NaturalNeighbours.AbstractInterpolator = NaturalNeighbours.Sibson{1}()
+    "Whether to use multithreading when interpolating.  Defaults to `nothing`, meaning that triangulation is multithreaded but interpolation is not.  May otherwise be `true` or `false`."
+    parallel::Union{Nothing, Bool} = nothing
+    "Whether to project the data onto the Delaunay triangulation.  Defaults to `true`."
+    project::Bool = true
+    "The method to use for the differentiation.  Defaults to `Direct()`.  May be `Direct()` or `Iterative()`."
+    derivative_method::NaturalNeighbours.AbstractDifferentiator = NaturalNeighbours.Direct()
+    "Whether to use cubic terms for estimating the second order derivatives. Only relevant for derivative_method == Direct()."
+    use_cubic_terms::Bool = true
+    "The weighting parameter used for estimating the second order derivatives. Only relevant for derivative_method == Iterative()."
+    alpha::Real = 0.1
+end
+
+function NaturalNeighboursMethod(method::NaturalNeighbours.AbstractInterpolator)
+    return NaturalNeighboursMethod(; method)
+end
+
+function (alg::NaturalNeighboursMethod)(
+        xrange::AbstractRange, yrange::AbstractRange,
+        positions::AbstractVector{<:Point{2}}, 
+        data::AbstractVector{<:Number}; 
+        mask=nothing
+        )
+    @assert length(positions) == length(data) "Positions (length $(length(positions))) and data (length $(length(data))) must have the same length."
+    # First, create the triangulated interpolator
+    interpolator = NaturalNeighbours.interpolate(
+        first.(positions), last.(positions), data; 
+        derivatives = true, method = alg.derivative_method, 
+        use_cubic_terms = alg.use_cubic_terms, alpha = alg.alpha,
+        parallel = isnothing(alg.parallel) ? true : alg.parallel
+    )
+    # Then, interpolate the data at the grid points.
+    nx, ny = length(xrange), length(yrange)
+    x_values = vec([x for x in xrange, _ in yrange])
+    y_values = vec([y for _ in xrange, y in yrange])
+    z_values = interpolator(
+        x_values, y_values; 
+        method = alg.method, 
+        parallel = isnothing(alg.parallel) ? false : alg.parallel,
```

Comment:
The parallel options seem to be at odds with each other. Above, the derivatives are estimated in parallel when constructing the interpolant, but here in the `interpolator` the interpolant is evaluated without multithreading by default